runconfig/hostconfig.go

```diff
@@ -2,11 +2,39 @@ package runconfig
 
 import (
 	"encoding/json"
+	"fmt"
 	"io"
 
 	"github.com/docker/docker/api/types/container"
 )
 
+var (
+	// ErrConflictContainerNetworkAndLinks conflict between --net=container and links
+	ErrConflictContainerNetworkAndLinks = fmt.Errorf("Conflicting options: container type network can't be used with links. This would result in undefined behavior")
```

Comment:
Just a minor nit, but I'd either move these to `network.go` or `errors.go` instead of `hostconfig.go`